### DISCUSSION TITLE
Installation doesn't work?

### DISCUSSION THREAD

**Discussion by TheDutchCoder** [dotansimha/graphql-code-generator#7680] [Q&A] [answered]
I'm trying to follow the installation docs, but I don't think they're correct?
https://www.graphql-code-generator.com/docs/getting-started/installation#initialization-wizard

Running `npm run graphql-codegen init` results in `Missing script: "graphql-codegen"`

Running it with npx results in a `MODULE_NOT_FOUND` error.

**Comment by ardatan** [comment, marked as answer]
Make sure you run `npm install graphql @graphql-codegen/cli` before. Otherwise please create a reproduction by following this template;
https://github.com/dotansimha/graphql-code-generator/issues/new?assignees=&labels=&template=bug_report.md

**Reply by TheDutchCoder** [reply]
I did, it still doesn't work. I only can get it to work with `npx graphql-codegen init` which is not directly in the docs (or at least it's ambiguous, as the docs say "The init process above can also be run through npx." which doesn't make it clear you need to drop the `run` part.

**Reply by bastianhilton** [reply]
agreed, the documentations need to mention that it should be npx graphql-codegen init as this worked for me. Thank you for the solution.

**Comment by GoPotatos** [comment]
Yeah there's a missing step that wasn't mentioned in the docs. You need to add the script to your package.json file

`"scripts": {    
    "graphql-codegen": "@graphql-codegen",
    /* Your other scripts */,
  },`

**Comment by igorek699669** [comment]
hello, in my case it is didnt work "npx graphql-code-generator init"  . I simply can't choose the answer - terminal throws me on a command line below

**Reply by jkmco** [reply]
yes me too, can u find the way to solve it?

**Reply by Daniel311099** [reply]
I know this isn't ideal but you could roll back to a previous version, I have the same issue and it works at least as recently as 1.21.6

**Reply by igorek699669** [reply]
thank you!

**Reply by igorek699669** [reply]
> I know this isn't ideal but you could roll back to a previous version, I have the same issue and it works at least as recently as 1.21.6

sorry but .. what module i need to lower version)?

**Reply by igorek699669** [reply]
and do i need to reset all back after init?

**Reply by nnnext** [reply]
You can change this in the package.json under devDependencies graphql-codegen/cli.... you can use 2.10.0 - this works for me at the moment but there seems to be a problem with the current version. fresh nextjs installation and i´m not able to run init.

**Reply by losandies** [reply]
I downgraded but still cant put in any options. sigh
